src/commonTest/kotlin/org/hisp/dhis/lib/expression/math/GS1ElementsTest.kt

```diff
@@ -26,4 +26,12 @@ internal class GS1ElementsTest {
         assertEquals(GS1Elements.PRODUCT_URL, GS1Elements.fromKey("PRODUCT URL"))
         assertEquals(GS1Elements.PRODUCT_URL, GS1Elements.fromKey("PRODUCTURL"))
     }
+
+    @Test
+    fun getApplicationIdentifierUsesActualCharacterForStar() {
+        assertEquals("250", GS1Elements.getApplicationIdentifier("250"))
+        assertEquals("314", GS1Elements.getApplicationIdentifier("314"))
```

Comment:
Actually, `314` should not be recognized as a valid application identifier. It's a quite harmless error, but it's going to look a bit strange if this unit test claiming it to be expected (hence correct) behavior. I would suggest either handling this case properly in the code or adding a comment to this particular test. Probably easier to fix the code than justifying not doing it in a comment though...

Reply:
Fixed it